### ISSUE TITLE
Uncaught Error: watch ~/Devel/atom/atom_configuration/toolbar.cson EMFILE

### ISSUE THREAD

**Issue by alflanagan** [cakecatz/flex-toolbar#150]
[Enter steps to reproduce:]

1. Started atom with a directory ('.') as the sole argument.
2. ????

[toolbar.cson.txt](https://github.com/cakecatz/flex-toolbar/files/2146905/toolbar.cson.txt)

**Atom**: 1.29.0-beta0 x64
**Electron**: 2.0.3
**OS**: "Fedora release
**Thrown From**: [flex-tool-bar](https://github.com/cakecatz/flex-toolbar) package 2.1.0

### Stack Trace

Uncaught Error: watch /home/aflanagan/Devel/atom/atom_configuration/toolbar.cson EMFILE

```
At events.js:183

Error: watch /home/aflanagan/Devel/atom/atom_configuration/toolbar.cson EMFILE
    at _errnoException (util.js:1024:11)
    at FSWatcher.start (fs.js:1386:19)
    at Object.fs.watch (fs.js:1412:11)
    at createFsWatchInstance (/packages/flex-tool-bar/node_modules/chokidar/lib/nodefs-handler.js:37:15)
    at setFsWatchListener (/packages/flex-tool-bar/node_modules/chokidar/lib/nodefs-handler.js:80:15)
    at FSWatcher.NodeFsHandler._watchWithNodeFs (/packages/flex-tool-bar/node_modules/chokidar/lib/nodefs-handler.js:229:14)
    at FSWatcher.NodeFsHandler._handleFile (/packages/flex-tool-bar/node_modules/chokidar/lib/nodefs-handler.js:256:21)
    at /packages/flex-tool-bar/node_modules/chokidar/lib/nodefs-handler.js:474:21)
    at FSReqWrap.oncomplete (fs.js:153:5)
```

### Commands

```
```

### Non-Core Packages

```
advanced-open-file 0.16.8 
atom-beautify 0.32.5 
atom-ide-debugger-node 0.7.1 
atom-ide-debugger-python 0.7.1 
atom-ide-ui 0.12.0 
atom-keyboard-macros 0.12.1 
atom-live-grammar-reload 0.8.0 
atom-notes 1.19.2 
autoclose-html 0.23.0 
autocomplete-paths 2.12.2 
autocomplete-python 1.10.5 
build 0.70.0 
build-cmake 0.11.0 
build-gradle 0.6.0 
build-make 0.13.0 
build-tsc 0.7.2 
busy-signal 1.4.3 
clipboard-plus 0.5.1 
disable-keybindings 0.6.0 
django-templates 0.5.0 
docblockr 0.13.7 
editorconfig 2.2.2 
emmet 2.4.3 
export-html 3.1.0 
file-icons 2.1.21 
flex-tool-bar 2.1.0 
go-plus 5.8.3 
grammar-mode 1.1.0 
Hydrogen 2.5.1 
ide-css 0.3.0 
ide-flowtype 0.23.0 
ide-go 0.7.0 
ide-html 0.4.2 
ide-json 0.2.1 
ide-php 0.7.14 
ide-typescript 0.7.6 
intentions 1.1.5 
jumpy 4.2.0 
language-babel 2.85.0 
language-generic-config 1.4.0 
language-ini 1.20.0 
language-markdown 0.26.0 
language-powershell 4.0.1 
language-svg 0.9.2 
language-utl 0.1.0 
last-cursor-position 0.9.2 
linter 2.2.0 
linter-coffeescript 1.0.0 
linter-gcc 0.7.1 
linter-js-standard 5.1.0 
linter-jsonlint 1.3.0 
linter-less 2.6.0 
linter-phpcs 1.6.8 
linter-pycodestyle 2.1.3 
linter-pydocstyle 0.4.7 
linter-shellcheck 1.4.7 
linter-tidy 2.3.1 
linter-ui-default 1.7.1 
linter-xmllint 1.4.4 
MagicPython 1.0.12 
minimap 4.29.8 
minimap-git-diff 4.3.1 
node-repl 0.5.0 
open-unsupported-files 1.0.20 
process-palette 0.16.1 
project-manager 3.3.6 
pure-syntax 0.3.0 
python-snippets 1.9.0 
python-yapf 0.14.0 
script 3.18.1 
sort-lines 0.18.0 
source-preview 0.5.3 
split-diff 1.5.2 
standard-formatter 2.8.0 
Sublime-Style-Column-Selection 1.7.4 
svg-preview 0.12.1 
tool-bar 1.1.7 
```

**Comment by UziTech** [comment]
According to google: [source](https://stackoverflow.com/a/10421706/806777)

> EMFILE error means that the OS is denying your program to open more files/sockets.
>
> Have a look at: [How do I change the number of open files limit in Linux?](https://stackoverflow.com/questions/34588/how-do-i-change-the-number-of-open-files-limit-in-linux)

**Comment by UziTech** [comment]
This might be fixed by flex-toolbar using [PathWatcher](https://atom.io/docs/api/v1.28.0/PathWatcher) instead of [chokidar](https://www.npmjs.com/package/chokidar)

**Comment by UziTech** [comment]
Is this a reproducible error or did it just happen once?

**Comment by alflanagan** [comment]
Sorry, haven't been able to reproduce recently as my whole Linux instance FUBAR'd. Pretty sure that wasn't caused by this issue. ;-)

**Comment by UziTech** [comment]
This might be fixed by v2.1.1

I'm going to close this issue. I can re-open it if it happens again.
